app/ui/browser-modern/sagas/page-sagas.js

```diff
@@ -77,6 +79,30 @@ export function* destroyPageSession({ id, withUI }) {
   }
 }
 
+function* forkPage(pageId, historyIndex) {
+  const currentPageIndex = yield select(PageSelectors.getPageIndexById, pageId);
+  const historyState = yield select(PageSelectors.getPageHistory, pageId);
+  const historyList = escape(JSON.stringify(historyState.toJS().map(history => history.url)));
+  const url = `tofino://sessionrestore/?history=${historyList}&historyIndex=${historyIndex}`;
+  const action = PageEffects.createPageSession(url, { selected: true });
+  const { id } = action;
+
+  yield createPageSession(action);
+  yield put(PageActions.setPageIndex(id, currentPageIndex + 1));
+}
+
+export function* forkPageBack({ pageId }) {
+  const historyIndex = yield select(PageSelectors.getPageHistoryIndex, pageId);
+  const newHistoryIndex = historyIndex - 1;
```

Comment:
I guess you can't fork a page back/forward when you can't navigate back and forward since the buttons are disabled. Would it be worth it to have an assertion here instead, or throw an error if new history indices are out of bounds.